pkg/tools/spire/start.go

```diff
@@ -220,3 +209,22 @@ func writeDefaultConfigFiles(ctx context.Context, spireRoot string) (string, err
 	}
 	return spireSocketName, nil
 }
+
+func execHealthCheck(ctx context.Context, cmdStr string, options ...*exechelper.Option) error {
+	if _, ok := ctx.Deadline(); !ok {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, healthCheckTimeout)
+		defer cancel()
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
```

Comment:
It looks like this branch could load CPU. Should we add wait in case if err != nil?

Reply:
`exechelper.Run(...)` starts process with `os/exec` and waits for its completion, so there is actually a wait, it is not a "for true" case.

Reply:
yes, but the process may fail in 50 ms

Reply:
We can, so why not :)